File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;
class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];

        
        char lastWord=words[0].charAt(0);
        
        HashSet<String> ms=new HashSet<String>();
    
        for (int i=0;i<words.length;i++){
            if (words[i].length()<=1){
                answer[0]=(i)%n+1;
                answer[1]=((i)/n)+1;
                return answer;
            }
            if (ms.contains(words[i])){
                answer[0]=(i)%n+1;
                answer[1]=((i)/n)+1;
                return answer;
            }
            if (words[i].charAt(0)!=lastWord){
                answer[0]=(i)%n+1;
                answer[1]=((i)/n)+1;
                return answer;
            }
            ms.add(words[i]);
            lastWord=words[i].charAt(words[i].length()-1);
        }

        return answer;
    }
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = [0,0]

    lastWord=words[0][0]
    ms=set()
    for i in range (len(words)):
        if lastWord!=words[i][0] or words[i] in ms:
            answer[0]=i%n+1
            answer[1]=i//n+1
            return answer
        ms.add(words[i])
        lastWord=words[i][len(words[i])-1]

    return answer